### ISSUE TITLE
​​False Positive in IL_INFINITE_LOOP Detector​​

### ISSUE THREAD

**Issue by RJerrica** [spotbugs/spotbugs#3755]
Hi, SpotBugs' IL_INFINITE_LOOP detector produced a false positive in the following code example by reporting an infinite loop warning at line 6. This is a false positive, as the program does terminate.

### Code Example
```java
class MethodCapitalization {
  public int hashcode() {
    return 42;
  }

  MethodCapitalization methodCapitalizationInstance = new MethodCapitalization(); // report a false INFINITE_LOOP warning
  int temp = methodCapitalizationInstance.hashcode();
}
```

**Comment by ThrawnCA** [comment]
Shouldn't the `methodCapitalizationInstance` field be `static`?
